src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -861,29 +873,21 @@ public override BoundNode VisitObjectCreationExpression(BoundObjectCreationExpre
             }
 
             var result = base.VisitObjectCreationExpression(node);
-
-            SetResultIsNotNull(node);
+            SetResultType(node);
 
             _implicitReceiver = saveImplicitReceiver;
             return result;
         }
 
-        private void SetResultIsNotNull(BoundExpression node)
+        private void SetResultType(BoundExpression node)
         {
             Debug.Assert(!IsConditionalState);
 
             // PROTOTYPE(NullableReferenceTypes): Is it necessary to check
             // this.State.Reachable for null checks, here or elsewhere?
-            if (this.State.Reachable)
+            //if (this.State.Reachable)
             {
-                if (node.Type?.IsReferenceType == true)
-                {
-                    this.State.ResultIsNotNull = true;
-                }
-                else
-                {
-                    this.State.ResultIsNotNull = null;
-                }
+                this.State.ResultType = TypeSymbolWithAnnotations.Create(node.Type);
```

Comment:
>TypeSymbolWithAnnotations.Create(node.Type); [](start = 40, length = 44)

It feels like nullability for nested type references captured in the bound tree might not be accurate/adjusted based on flow analysis.

Reply:
The nested nullability of the type in the bound tree does not account for flow analysis. I'm working on that now.

Reply:
> The nested nullability of the type in the bound tree does not account for flow analysis. I'm working on that now.

Do you recommend waiting for that change, or continuing with review?

Reply:
It probably makes sense to wait for that change, thanks.